File: solutions/python/armstrong-numbers/1/armstrong_numbers.py
def is_armstrong_number(number):
    str_number = str(number)
    total = 0
    for num in str_number:
        total += int(num)**len(str_number)
    return total == number


"""def is_armstrong_number(number):
    str_number = str(number)
    return sum(int(digit) ** len(str_number) for digit in str_number) == number"""

File: solutions/java/armstrong-numbers/1/src/main/java/ArmstrongNumbers.java
class ArmstrongNumbers {

    boolean isArmstrongNumber(int numberToCheck) {
        int total = 0;
        int number = numberToCheck;
        int i = (int) Math.floor(Math.log10(numberToCheck)) + 1;
        while (number > 0) {
            total += (int) Math.pow((number % 10), i);
            number = number / 10;
        }
        return total == numberToCheck ? true : false;

    }

}
